iroha/src/crypto.rs

```diff
@@ -106,6 +107,66 @@ impl Debug for Signature {
     }
 }
 
+/// Describes an object that can be signed.
+pub trait Sign {
+    /// The bytes that are signed.
+    fn payload(&self) -> Vec<u8>;
+
+    /// Method to get the signature based on the `payload`.
+    fn signature(
+        &self,
+        public_key: &PublicKey,
+        private_key: &PrivateKey,
+    ) -> Result<Signature, String> {
+        Ok(Signature::new(*public_key, &self.payload(), private_key)?)
+    }
+}
```

Comment:
Looks like we do not need a trait for `Sign`, but for objects which contains `payload`. After that we can pass them to the `Signature`'s constructor.
But because payload is just a vector of bytes, I don't think we need to provide a new trait at all.